src/syntax_analysis.rs

```diff
@@ -0,0 +1,192 @@
+use std::collections::HashMap;
+use syntect::parsing::syntax_definition::{
+    ContextReference, MatchOperation, MatchPattern, Pattern, SyntaxDefinition,
+};
+use syntect::parsing::{Scope, SyntaxSet, SyntaxSetBuilder};
+
+/// Used to look up (by name) what dependencies a given [SyntaxDefinition] has
+type SyntaxToDependencies = HashMap<String, Vec<Dependency>>;
+
+/// Used to look up what [SyntaxDefinition] that corresponds to a given [Dependency]
```

Comment:
```suggestion
/// Used to look up which [SyntaxDefinition] corresponds to a given [Dependency]
```
(if I'm interpreting the sentence correctly)